node/src/reactor/main_reactor/keep_up.rs

```diff
@@ -344,39 +308,37 @@ impl MainReactor {
     }
 
-    // Attempts to read the validators from the global states of the immediate switch block and its
+    // Attempts to read the validators from the global states of the block after the upgrade and its
     // parent; initiates fetching of the missing global states, if any.
-    fn try_read_validators_for_immediate_switch_block(
+    fn try_read_validators_for_block_after_upgrade(
         &mut self,
         effect_builder: EffectBuilder<MainEvent>,
-        block_hash: BlockHash,
-        block_era_id: EraId,
-        parent_metadata: ParentMetadata,
+        global_states_metadata: GlobalStatesMetadata,
     ) -> KeepUpInstruction {
         // We try to read the validator sets from global states of two blocks - if either returns
         // `RootNotFound`, we'll initiate fetching of the corresponding global state.
         let effects = async move {
             // Send the requests to contract runtime.
-            let parent_era_validators_request = EraValidatorsRequest::new(
-                parent_metadata.parent_state_hash,
-                parent_metadata.parent_protocol_version,
+            let before_era_validators_request = EraValidatorsRequest::new(
+                global_states_metadata.before_state_hash,
+                global_states_metadata.before_protocol_version,
             );
-            let parent_era_validators_result = effect_builder
-                .get_era_validators_from_contract_runtime(parent_era_validators_request)
+            let before_era_validators_result = effect_builder
+                .get_era_validators_from_contract_runtime(before_era_validators_request)
                 .await;
-            let block_era_validators_request = EraValidatorsRequest::new(
-                parent_metadata.global_state_hash,
-                parent_metadata.protocol_version,
+            let after_era_validators_request = EraValidatorsRequest::new(
+                global_states_metadata.after_state_hash,
+                global_states_metadata.after_protocol_version,
             );
-            let block_era_validators_result = effect_builder
-                .get_era_validators_from_contract_runtime(block_era_validators_request)
+            let after_era_validators_result = effect_builder
+                .get_era_validators_from_contract_runtime(after_era_validators_request)
                 .await;
 
             // Check the results.
             // A return value of `Ok` means that validators were read successfully.
             // An `Err` will contain a vector of (block_hash, global_state_hash) pairs to be
             // fetched by the `GlobalStateSynchronizer`, along with a vector of peers to ask.
-            match (parent_era_validators_result, block_era_validators_result) {
+            match (before_era_validators_result, after_era_validators_result) {
                 // Both states were present - return the result.
                 (Ok(parent_era_validators), Ok(block_era_validators)) => {
```

Comment:
Nit for consistency: consider renaming:
`parent_era_validators` ==> `before_era_validators`
`block_era_validators` ==> `after_era_validators`

Reply:
Good catch! I must have missed it, I completely agree with your suggestion.

Reply:
Addresses in 2bd2566